### PR TITLE
Add a walking skeleton for the halight-based swaps

### PR DESCRIPTION
**Pull request by thomaseizinger**
To make the skeleton walk, we opted to only support swaps from han-ethereum-ether to halight-lightning-bitcoin.

Resolves https://github.com/comit-network/comit-rs/issues/2297.

PR https://github.com/comit-network/comit-rs/pull/2383 should be merged first because it contains the same changes.

### CODE REVIEW

File: cnd/Cargo.toml [hunk 1]
```diff
@@ -35,6 +36,7 @@ libp2p-comit = { path = "../libp2p-comit" }
 libsqlite3-sys = { version = ">=0.8.0, <0.13.0", features = ["bundled"] }
 log = { version = "0.4", features = ["serde"] }
 lru = "0.4.3"
+multihash = "0.10" # TODO: Upgrade libp2p and use libp2p::multihash instead
```
Comment (thomaseizinger) [inline cnd/Cargo.toml:39]: :mega: This TODO needs to be removed.

Comment (tcharding) [reply cnd/Cargo.toml:39]: Man I deleted that little sucker about 10 times during the `ln-single-patch` branch - how did it get back in :)

File: cnd/src/swap_protocols/han.rs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+/// The current implementation is naive in the sense that it does not take into
+/// account situations where it is clear that no more events will happen even
+/// though in theory, there could. For example:
+/// - funded
+/// - refunded
+///
+/// It is highly unlikely for Bob to fund the HTLC now, yet the current
+/// implementation is still waiting for that.
+pub async fn create_watcher<C, S, L, A, H, I, T>(
+    ethereum_connector: &C,
```
Comment (da-kami) [inline cnd/src/swap_protocols/han.rs:31]: ❓ Is this still restricted to Ethereum only?

File: cnd/src/swap_protocols/han.rs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            }
+        }
+    }
+}
+
+/// Returns a future that waits for events to happen on a ledger.
+///
+/// Each event is yielded through the controller handle (co) of the coroutine.
+async fn watch_ledger<C, L, A, H, I, T>(
+    ethereum_connector: &C,
```
Comment (da-kami) [inline cnd/src/swap_protocols/han.rs:88]: ❓  Is this still restricted to Ethereum only? (looks like already generic to me)

File: cnd/src/swap_protocols/han.rs (hunk 2 above)
Comment (thomaseizinger) [reply cnd/src/swap_protocols/han.rs:31]: To be fixed :)

File: api_tests/src/actors/actor.ts [hunk 4]
```diff
@@ -277,12 +354,32 @@ export class Actor {
         }
     }
 
+    public async init() {
+        if (!this.swap) {
+            throw new Error("Cannot init nonexistent swap");
+        }
+
+        const response = await this.swap.tryExecuteSirenAction<LedgerAction>(
+            "init",
```
Comment (D4nte) [inline api_tests/src/actors/actor.ts:363]: I don't think this is a great name. I find it confusing despite being well versed with halight.

Comment (thomaseizinger) [reply api_tests/src/actors/actor.ts:363]: That is inconsistent with our other action naming though. The name of the action describes the phase and not the actual action that is going to be executed. I see `deploy` being replaced with `init` eventually.

File: api_tests/src/test_environment.ts [hunk 5]
```diff
@@ -259,7 +259,12 @@ export default class TestEnvironment extends NodeEnvironment {
     private async setupLightningChannels() {
         const { alice, bob } = this.global.lndWallets;
 
-        await alice.connectPeer(bob);
+        const alicePeers = await alice.listPeers();
+        const bobPubkey = await bob.inner.getPubkey();
+
+        if (!alicePeers.find((peer) => peer.pubKey === bobPubkey)) {
```
Comment (thomaseizinger) [inline api_tests/src/test_environment.ts:265]: Could have done this check inside the fn to make it idempotent :)

Anyway, thanks for the fix! <3

File: .circleci/config.yml [hunk 6]
```diff
@@ -94,15 +94,15 @@ commands:
   install_lnd:
     steps:
       - run:
-          name:  "Install go 1.13 & lnd"
+          name:  "Install go 1.13 & lnd v0.9.10-beta"
```
Comment (thomaseizinger) [inline .circleci/config.yml:97]:
```suggestion
          name:  "Install go 1.13 & lnd v0.9.1-beta"
```

File: api_tests/tests/ether_halight.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import { twoActorTest } from "../src/actor_test";
+import SwapFactory from "../src/actors/swap_factory";
+import { sleep } from "../src/utils";
+
+it(
+    "han-ethereum-ether-halight-lightning-bitcoin-alice-redeems-bob-redeems",
+    twoActorTest(async ({ alice, bob }) => {
+        const [aliceBody, bobBody] = await SwapFactory.newSwap(alice, bob);
+
+        // make sure bob knows about the swap first
```
Comment (D4nte) [inline api_tests/tests/ether_halight.ts:15]: Why? Can the why be explain in the comment please @thomaseizinger

Comment (thomaseizinger) [reply api_tests/tests/ether_halight.ts:15]: Can do yes!

**Comment by D4nte** [comment]
@thomaseizinger let's stop pushing on the branch and see if it passes? :)

**Comment by thomaseizinger** [comment]
> @thomaseizinger let's stop pushing on the branch and see if it passes? :)

I passed in GH actions already before I pushed!

**Comment by thomaseizinger** [comment]
@D4nte https://github.com/comit-network/comit-rs/pull/2426/checks?check_run_id=573084877

**Comment by D4nte** [comment]
> > @thomaseizinger let's stop pushing on the branch and see if it passes? :)
> 
> I passed in GH actions already before I pushed!

Awesome 👍🎉 Let's do it bors!!!

**Comment by thomaseizinger** [comment]
> > > @thomaseizinger let's stop pushing on the branch and see if it passes? :)
> > 
> > 
> > I passed in GH actions already before I pushed!
> 
> Awesome +1tada Let's do it bors!!!

But I want to clean up the commits XD

**Comment by thomaseizinger** [comment]
Commits cleaned up! Ready to go!

**Comment by D4nte** [comment]
bors retry

**Pull request merged**